libs/dyn/merge/merge.go

```diff
@@ -88,11 +92,19 @@ func mergeSequence(a, b dyn.Value) (dyn.Value, error) {
 	copy(out[:], as)
 	copy(out[len(as):], bs)
 
-	// Preserve the location of the first value.
-	return dyn.NewValue(out, a.Location()), nil
+	// Preserve the location of the first value. Accumulate the locations of the second value.
+	return mergeLocations(dyn.NewValue(out, a.Locations()), b), nil
```

Comment:
For maps and sequences we currently always return the location where it first showed up.

Is this still the case now?

Reply:
Yeah it is. Should be clearer now with the refactor.